app/models/adapters/payment_history_adapter.rb

```diff
@@ -5,13 +5,18 @@ class PaymentHistoryAdapter
     attr_accessor :payments, :return_payments
 
     def initialize(input)
-      @input_payments = [input[:payments][:payment]].flatten
-      payments, return_payments = @input_payments.partition do |payment|
-        payment.dig(:return_payment, :check_trace_number).blank?
-      end
+      @payments = []
+      @return_payments = []
+
+      if input
+        @input_payments = [input[:payments][:payment]].flatten
```

Comment:
Is there any reason to think input might not have payments? That could lead to an error here.